utils/dynstream/stream.go

```diff
@@ -316,28 +323,26 @@ type pathInfo[A Area, P Path, T Event, D Dest, H Handler[A, P, T, D]] struct {
 	// guaranteed to see the memory change of this field.
 	removed atomic.Bool
 	// The path is blocked by the handler.
-	blocking bool
+	blocking atomic.Bool
```

Comment:
Why change it to an atomic variable? Atomic variable means more complex usage and is more error prone. So I want to know why it is changed in this pr.